### PR TITLE
Misc fix and improvements for the DR channel

### PR DESCRIPTION
**Pull request by hardening**

### CODE REVIEW

File: channels/drive/client/drive_main.c [hunk 1]
```diff
@@ -592,13 +614,16 @@ static UINT drive_process_irp_query_directory(DRIVE_DEVICE* drive, IRP* irp)
 	if (!drive || !irp || !irp->Complete)
 		return ERROR_INVALID_PARAMETER;
 
+	if (Stream_GetRemainingLength(irp->input) < 28)
```
Comment (bmiklautz) [inline channels/drive/client/drive_main.c]: Confused here: UINT32+UINT8+UINT32+23*UINT8 isn't that 32?

File: channels/drive/client/drive_main.c [hunk 2]
```diff
@@ -274,10 +274,13 @@ static UINT drive_process_irp_read(DRIVE_DEVICE* drive, IRP* irp)
 	if (!drive || !irp || !irp->output || !irp->Complete)
 		return ERROR_INVALID_PARAMETER;
 
+	if (Stream_GetRemainingLength(irp->input) < 6)
```
Comment (bmiklautz) [inline channels/drive/client/drive_main.c]: I think that should be < 12

File: channels/drive/client/drive_main.c [hunk 3]
```diff
@@ -328,11 +331,14 @@ static UINT drive_process_irp_write(DRIVE_DEVICE* drive, IRP* irp)
 	if (!drive || !irp || !irp->input || !irp->output || !irp->Complete)
 		return ERROR_INVALID_PARAMETER;
 
+	if (Stream_GetRemainingLength(irp->input) < 26)
```
Comment (bmiklautz) [inline channels/drive/client/drive_main.c]: Shouldn't that be < 32?

File: channels/drive/client/drive_main.c (hunk 2 above)
Comment (hardening) [reply channels/drive/client/drive_main.c]: yes of course !

File: channels/drive/client/drive_main.c (hunk 3 above)
Comment (hardening) [reply channels/drive/client/drive_main.c]: yep

File: channels/drive/client/drive_main.c (hunk 1 above)
Comment (hardening) [reply channels/drive/client/drive_main.c]: You're right again

**Comment by hardening** [comment]
So I've fixed all my wrong computations (looks like late patch aren't a so nice idea)

File: channels/drive/client/drive_main.c (hunk 2 above)
Comment (akallabeth) [reply channels/drive/client/drive_main.c]: @hardening Sure you don't want to change that too?

**Comment by hardening** [comment]
@bmiklautz @akallabeth I think I've fixed everything

**Comment by bmiklautz** [comment]
@giox069 could you re-test the current pull request if it's working for you?

**Comment by bmiklautz** [comment]
@hardening nice work!

**Comment by giox069** [comment]
I just tested it with the same couple of win2003 servers and a win10 server: I copied some files from the remote desktop to \\tsclient\shareddir without problems: it worked with no crashes and no file corruptions.

**Comment by bmiklautz** [comment]
@giox069 thanks.

**Pull request merged**